jerry-core/ecma/operations/ecma-typedarray-object.c

```diff
@@ -303,6 +312,12 @@ ecma_typedarray_create_object_with_typedarray (ecma_object_t *typedarray_p, /**<
   else
   {
     ecma_length_t byte_length = array_length << element_size_shift;
+
```

Comment:
Yes, I should check every shift operation. A big project.
what about using `array_length>>(sizeof (array_length) - shift) ==0` to indicate there is no overflow